source/connection.c

```diff
@@ -385,8 +432,15 @@ struct aws_http_server *aws_http_server_new(const struct aws_http_server_options
     server->initial_window_size = options->initial_window_size;
     server->user_data = options->server_user_data;
     server->on_incoming_connection = options->on_incoming_connection;
-
-    int err = aws_hash_table_init(
+    server->on_destroy_complete = options->on_destroy_complete;
+    server->synced_data.is_shutting_down = false;
```

Comment:
trivial/style: you don't need to change this, but we usually don't bother setting values that are 0/false at init time. Since we pretty much always grab the memory with calloc() (always zeroes out the memory). Or in cases where the memory already exists, we zero it out with AWS_STRUCT_ZERO() before setting up the fields